dot/state/block.go

```diff
@@ -705,6 +710,11 @@ func (bs *BlockState) HandleRuntimeChanges(newState *rtstorage.TrieState, rt run
 		return fmt.Errorf("failed to update code substituted block hash: %w", err)
 	}
 
+	newVersion, err := rt.Version()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve runtime version: %w", err)
+	}
+	bs.notifyRuntimeUpdated(newVersion)
```

Comment:
I wonder if it's good to put this in a goroutine since it might block if some channel is full

Reply:
good idea, added as goroutine.